app/src/main/java/dev/animeshvarma/sigil/ui/screens/CustomEncryptionScreen.kt

```
@@ -306,6 +374,341 @@ fun CustomEncryptionScreen(viewModel: SigilViewModel, uiState: UiState) {
+ * @param initialDescription The starting description for the profile.
+ * @param initialKdfOverride The existing KDF override if any, to prefill the toggle and sliders.
+ * @param defaultKdf The default/global KDF config to fallback to if no override is set.
+ * @param initialIsRaw The starting raw mode state.
+ * @param layerCount Number of layers in the current profile; controls whether the Raw Mode option is shown.
+ * @param onDismiss Callback invoked when the user cancels or dismisses the dialog.
+ * @param onSave Callback invoked when the user confirms the save. Receives the profile name, short description,
+ *               an optional `CryptoEngine.KdfConfig` (null when KDF override is not used), and a Boolean indicating
+ *               whether Raw Mode is enabled.
+ */
+@Composable
+fun SaveProfileDialog(
```

detekt (1.23.8)

[warning] The function SaveProfileDialog appears to be too complex based on Cognitive Complexity (complexity: 28). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)